protopipe/pipeline/event_preparer.py

```diff
@@ -436,12 +439,18 @@ def prepare_event(self, source, return_stub=False, save_images=False, debug=Fals
                         # to make Hillas parametrization faster
                         camera_biggest = camera[mask_reco]
                         image_biggest = image_biggest[mask_reco]
+                        if save_images is True:
+                        	dl1_phe_image_mask_reco[tel_id] = 1*mask_reco
+                   
                     elif num_islands > 1:  # if more islands survived..
                         # ...find the biggest one
                         mask_biggest = largest_island(labels)
                         # and also reduce dimensions
                         camera_biggest = camera[mask_biggest]
                         image_biggest = image_biggest[mask_biggest]
+                        if save_images is True:
+                        	dl1_phe_image_mask_reco[tel_id] = 1*mask_biggest
```

Comment:
Thank you for the review @HealthyPear.
The reason I've done like this is that I get an error opening the new images.h5 file with the boolean masks (is only me?).
So I've done like this:

> import numpy as np
image = np.array([27,2,13])

> mask = 1*np.array([ True, False, True]) 

> image*mask
array([27,  0, 13])

...it's really a dirty way to avoid the problem but in this way I was able to read the file and the output seems to me reasonable:
![example](https://user-images.githubusercontent.com/51763341/75171030-4e793d00-572b-11ea-802d-ad22069f4030.png)

Maybe I didn't use the correct boolean equivalent of tb.Float32Col(): as a first attempt I've used tb.BoolCol().

Reply:
Ah, I see.

Ok, so doing `image * mask` with the mask as you did will work also.
The problem is that if anyone else wants to use this feature, he/she will have to either convert all the arrays back to boolean, or change their code to clean the images in this other way.
Also, we should try to keep it coherent with the approach used in _ctapipe/protopipe_, which use boolean masks.

I am not an expert of PyTables - let's investigate this problem: could you please post the error you get when you try to use a boolean mask?

Reply:
Also, in the plots like the one you show here, perhaps highlighting the pixels to be saved in the left panel will help you check better by eye what happens

Reply:
Yes @HealthyPear I totally agree: would be better to keep the approach coherent.

The error is this one:

> TypeError: No NumPy equivalent for TypeBitfieldID exists

and I think that it's the same mentioned here https://github.com/astropy/astropy/issues/6473
...anyway I think I'll modify the code in order to save the boolean mask: then the images.h5 hasn't to be open with h5py as I did.

For the image here it is:
![example_2](https://user-images.githubusercontent.com/51763341/75373322-62a77080-58ca-11ea-91c1-b1aef548b810.png)

Reply:
> Yes @HealthyPear I totally agree: it would be better to keep the approach coherent.
> 
> The error is this one:
> 
> > TypeError: No NumPy equivalent for TypeBitfieldID exists
> 
> and I think that it's the same mentioned here [astropy/astropy#6473](https://github.com/astropy/astropy/issues/6473)

Yes, it seems indeed a problem with the h5py package.

> ...anyway I think I'll modify the code in order to save the boolean mask: then the images.h5 hasn't to be open with h5py as I did.

For what is worth, in _protopipe_ the notebooks use _PyTables_ to open the HDF5 files exported by the scripts!